client/state/data-getters/index.js

```diff
@@ -23,6 +23,31 @@ import {
 import { convertToSnakeCase } from 'state/data-layer/utils';
 import { dummyTaxRate } from 'lib/tax'; // #tax-on-checkout-placeholder
 
+/**
+ * Fetches content from a URL with a GET request
+ *
+ * The ID here is obscured and so **this should not
+ * be used inside a React component** but rather it
+ * should only be used in something like `waitForData()`
+ * where the wrapper defines the shape of the output
+ *
+ * @example
+ * waitForData( {
+ *     planets: getAtUrl( 'https://swapi.co/api/planets/' ),
+ * } ).then( ( { planets } ) => {
+ *     console.log( planets.data );
+ * } );
+ *
+ * @see waitForData
+ *
+ * @param {string} url location from which to GET data
+ * @return {object} HTTP data wrapped value
+ */
+export const getAtUrl = url =>
+	requestHttpData( `get-at-url-${ url }`, rawHttp( { method: 'GET', url } ), {
+		fromApi: () => data => [ `get-at-url-${ url }`, data ],
+	} );
```

Comment:
thanks for the review! I see where you are getting at and yeah, I don't know why components couldn't use this, on further reflection, though I'd suggest they wouldn't want to use `getHttpData()` if they really want the data.

I like the name `requestAtUrl` - I was thinking more narrowly about the problem at hand when writing this but `request` is a better fit.

is it a framework-ish thing? I don't know; I hope not. this is just a helper around the (poorly-named) http-data system which is actually agnostic about how it fetches data. that is, a keen eye might note that there is nothing about HTTP in the httpData system - it's all in Calypso's Redux actions in the data layer proper.

how do you see it as being more at the framework level?

Reply:
I remember also now that the original reason I called it `getAtUrl` was because it made a `GET`, but then that overlaps with our normal use of `get`

#namingishard

Reply:
The nittiest nit, but we can also drop the `At`. I think `requestUrl` or `getUrl` sound even better.

Reply:
> is it a framework-ish thing?

One way to test for frameworkishness is to ask: is the API generally useful for any app that uses the Calypso Framework? Or is it domain specific to the Calypso App? I'd say that almost all apps want to send out HTTP GET requests.

This question is definitely not a blocker for merging this PR. But one day we should reconsider about how our `client/state` and `client/state/data-layer` directories are organized. They are a mix of framework things that could be published as a Lerna package (various Redux helpers for state persistence, generic Data Layer things like the `dispatchRequest` machine, the request-processing pipeline, the http-data cache...) and Calypso-specific code like the handlers for fetching Privacy Policy etc.

> (poorly-named) http-data system which is actually agnostic about how it fetches data

React has a small library similar to http-data called `react-cache`. And Apollo GraphQL also has a component called "cache". When we start looking for a good name for http-data, strings matching `/cache/` will be good candidates.

Reply:
thanks for sharing those thoughts!

also on the name I have internally (in my head) already renamed it to `js-tasks` but since I don't actively develop it it's true name should probably be `neglected.js`